buildSrc/src/main/java/com/google/firebase/gradle/plugins/VendorPlugin.kt

```diff
@@ -0,0 +1,260 @@
+
+        val android = project.extensions.getByType(LibraryExtension::class.java)
+
+        android.registerTransform(VendorTransform(
+                android,
+                vendor,
+                JarJarTransformer(
+                        parentPackageProvider = {
+                            android.libraryVariants.find { it.name == "release" }!!.applicationId
+                        },
+                        jarJarProvider = { jarJar.resolve() },
+                        project = project,
+                        logger = project.logger),
+                logger = project.logger))
+    }
+}
+
+interface JarTransformer {
+    fun transform(inputJar: File, outputJar: File, packagesToVendor: Set<String>)
+}
+
+class JarJarTransformer(
+    private val parentPackageProvider: () -> String,
+    private val jarJarProvider: () -> Collection<File>,
+    private val project: Project,
+    private val logger: Logger
+) : JarTransformer {
+    override fun transform(inputJar: File, outputJar: File, packagesToVendor: Set<String>) {
+        val parentPackage = parentPackageProvider()
+        val rulesFile = File.createTempFile(parentPackage, ".jarjar")
+        rulesFile.printWriter().use {
+            for (externalPackageName in packagesToVendor) {
+                it.println("rule $externalPackageName.** $parentPackage.@0")
+            }
+        }
+        logger.info("The following JarJar configuration will be used:\n ${rulesFile.readText()}")
+
+        project.javaexec {
+            main = "org.pantsbuild.jarjar.Main"
+            classpath = project.files(jarJarProvider())
+            // jvmArgs = listOf("-Xdebug", "-Xrunjdwp:transport=dt_socket,server=y,suspend=y,address=5005")
+            args = listOf("process", rulesFile.absolutePath, inputJar.absolutePath, outputJar.absolutePath)
+            systemProperties = mapOf("verbose" to "true", "misplacedClassStrategy" to "FATAL")
+        }.assertNormalExitValue()
+    }
+}
+
+class VendorTransform(
+    private val android: LibraryExtension,
+    private val configuration: Configuration,
+    private val jarTransformer: JarTransformer,
+    private val logger: Logger
+) :
+        Transform() {
+    override fun getName() = "firebaseVendorTransform"
+
+    override fun getInputTypes(): MutableSet<QualifiedContent.ContentType> {
+        return mutableSetOf(QualifiedContent.DefaultContentType.CLASSES)
+    }
+
+    override fun isIncremental() = false
+
+    override fun getScopes(): MutableSet<in QualifiedContent.Scope> {
+        return mutableSetOf(QualifiedContent.Scope.PROJECT)
+    }
+
+    override fun getReferencedScopes(): MutableSet<in QualifiedContent.Scope> {
+        return mutableSetOf(QualifiedContent.Scope.PROJECT)
+    }
+
+    override fun transform(transformInvocation: TransformInvocation) {
+        if (configuration.resolve().isEmpty()) {
+            logger.info("Nothing to vendor. " +
```

Comment:
maybe upgrade it to warning? I think it's more likely that devs forgot to include the lib to vendor than that they forgot to remove the unnecessary plugin.